### ISSUE TITLE
File creation fails silently when filename is longer than about 255 bytes

### ISSUE THREAD

**Issue by evandjohnston** [yausername/dvd#18]
### tl;dr

Most filesystems support filenames of ≤255 bytes, but youtube-dl sometimes generates filenames longer than 255 bytes. When this happens, file creation fails silently. Suggest truncating filenames longer than 255 bytes.

### Detail

When trying to download certain videos, dvd successfully loads the metadata, allows selection of download location, and displays "Download queued" toast popup, however, the video doesn't appear in the Downloads tab nor on the filesystem.

I created a [Twitter thread of testcases](https://twitter.com/evandjohnston/status/1368658344844562434) which suggests that the issue is due to filename length. It seems [most filesystems have a max filename length of 255 bytes](https://askubuntu.com/questions/859945/what-is-the-maximum-length-of-a-file-path-in-ubuntu), so my guess is that dvd is running into that cap.

It seems like the fix could be as simple as truncating the filename. Catching failed downloads and notifying the user would also be a nice-to-have, but would be more work to implement. 😉

Thanks for your work on this project!

**Comment by yausername** [comment]
Hey,

Thanks for the detailed bug report :)
Will push a fix when I find some time (which is hard lately).

**Comment by uomar8** [comment]
> Hey,
> 
> Thanks for the detailed bug report :)
> Will push a fix when I find some time (which is hard lately).


I found a solution
Change

-o /%(title)s.%(ext)s 

To

/%(title).200s.%(ext)s

This will limit title to 200 character

**Comment by suzukar** [comment]
> ### tl;dr
> Most filesystems support filenames of ≤255 bytes, but youtube-dl sometimes generates filenames longer than 255 bytes. When this happens, file creation fails silently. Suggest truncating filenames longer than 255 bytes.
> 
> ### Detail
> When trying to download certain videos, dvd successfully loads the metadata, allows selection of download location, and displays "Download queued" toast popup, however, the video doesn't appear in the Downloads tab nor on the filesystem.
> 
> I created a [Twitter thread of testcases](https://twitter.com/evandjohnston/status/1368658344844562434) which suggests that the issue is due to filename length. It seems [most filesystems have a max filename length of 255 bytes](https://askubuntu.com/questions/859945/what-is-the-maximum-length-of-a-file-path-in-ubuntu), so my guess is that dvd is running into that cap.
> 
> It seems like the fix could be as simple as truncating the filename. Catching failed downloads and notifying the user would also be a nice-to-have, but would be more work to implement. 😉
> 
> Thanks for your work on this project!

I also encountered the same problem.
